eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/contig/ContigMapping.java

```diff
@@ -20,50 +20,114 @@
 import uk.ac.ebi.eva.accession.dbsnp.io.AssemblyReportReader;
 
 import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
+import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class ContigMapping {
 
-    private Map<String, ContigSynonyms> sequenceNameToSynonyms;
+    private Map<String, ContigSynonyms> sequenceNameToSynonyms = new HashMap<>();
 
-    private Map<String, ContigSynonyms> genBankToSynonyms;
+    private Map<String, ContigSynonyms> assignedMoleculeToSynonyms = new HashMap<>();
 
-    private Map<String, ContigSynonyms> refSeqToSynonyms;
+    private Set<String> assignedMoleculeDuplicates = new HashSet<>();
 
-    private Map<String, ContigSynonyms> ucscToSynonyms;
+    private Map<String, ContigSynonyms> genBankToSynonyms = new HashMap<>();
+
+    private Map<String, ContigSynonyms> refSeqToSynonyms = new HashMap<>();
+
+    private Map<String, ContigSynonyms> ucscToSynonyms = new HashMap<>();
 
     private static final String CHROMOSOME_PATTERN = "(chromosome|chrom|chr|ch)(.+)";
 
     private static final Pattern PATTERN = Pattern.compile(CHROMOSOME_PATTERN, Pattern.CASE_INSENSITIVE);
 
+    private static final String NOT_AVAILABLE = "na";
+
     public ContigMapping(String assemblyReportUrl) throws Exception {
         this(new AssemblyReportReader(assemblyReportUrl));
     }
 
     public ContigMapping(AssemblyReportReader assemblyReportReader) throws Exception {
-        sequenceNameToSynonyms = new HashMap<>();
-        genBankToSynonyms = new HashMap<>();
-        refSeqToSynonyms = new HashMap<>();
-        ucscToSynonyms = new HashMap<>();
         populateMaps(assemblyReportReader);
     }
 
+    public ContigMapping(List<ContigSynonyms> contigSynonyms) {
+        contigSynonyms.forEach(this::fillContigConventionMaps);
```

Comment:
I don't find the suggested name particularly clear (what is it convenient for?) and would prefer to keep the current one.

Reply:
I thought they were "convenience" maps because they were convenient for retrieving the contig synonyms using different kinds of accessions, and that the name was misspelled. But if the meaning is other, then my comment should be ignored.

Reply:
they are some kind of convenience maps as they make easier the search. we could have had `List<ContigSynonym>` and make dumb searches, but that would be very slow. Each map allows searching by a different convention (genbank, refseq, seqname, etc).